discojs/discojs-core/src/training/disco.ts

```diff
@@ -56,26 +47,26 @@ export class Disco {
         options.url = new URL(options.url)
       }
       switch (options.scheme) {
-        case TrainingSchemes.FEDERATED:
+        case 'federated':
           options.client = new clients.federated.FederatedClient(options.url, task, options.aggregator)
           break
-        case TrainingSchemes.DECENTRALIZED:
+        case 'decentralized':
           options.client = new clients.decentralized.DecentralizedClient(options.url, task, options.aggregator)
           break
-        default:
+        case 'local':
           options.client = new clients.Local(options.url, task, options.aggregator)
           break
       }
```

Comment:
Should we add a fallback error?

Reply:
in the potential `default` case, the type of `options.scheme` is actually `never`, thanks to [exhaustiveness checking](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#exhaustiveness-checking) there is no way that it'll happen (iff types are correct), so it'd be dead code.
but yeah, let's add a compile time error to ensure that's it's indeed not happening (might be the case when changing types)